packages/task/src/browser/task-name-resolver.ts

```diff
@@ -33,10 +33,10 @@ export class TaskNameResolver {
      */
     resolve(task: TaskConfiguration): string {
         if (this.isDetectedTask(task)) {
-            const scope = task._scope;
+            const scope = task.scope;
```

Comment:
Is the idea here to substitute `scope` for `_scope` throughout the detected task system? Will that have any side effects on other code that looks for the `_scope` field?

Reply:
task._scope can change to reflect the Uri of the root where it's being applied, however task.scope remains the same as submitted by the provider, so using task.scope help us correlate configured tasks against provided tasks which would not be possible if we use the updated task._scope. 
  I don't actually want to expand this approach to other places, and so far I don't see any impacts to other areas.
Please let me know if you see potential side effects.

Reply:
So as long as it's coming from the `detected` task system, it'll have the property `scope`, and if the user chooses to add it to the configured task system, it'll have a `._scope`, basically? It looks like the respective handlers are looking for the right thing, so that sounds good.

Reply:
Yes, that's my understanding as well,

Reply:
> as long as it's coming from the detected task system, it'll have the property scope, and if the user chooses to add it to the configured task system, it'll have a ._scope

I think it makes sense to add such description somewhere in code, sure not right here, but to the corresponding fields of an interface. I think it should be really useful for people who work with tasks related area.
wdyt?